### PR TITLE
PI: import options UI

### LINKED ISSUE #2452: Product import: set up dropdown selector to decide where you want to import
## Description
As an enterprise user, when I start the product import process, I am asked to choose where I want to import the info I am going to upload: product listing or inventory.

## Acceptance criteria
capture du 2018-07-12 00-10-01

- see a dropdown menu and I am asked to choose
- depending on the case I choose the import work as expected in the given case

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #2452 and #2460.

Simplifying Product Import UI. Removed most of the import options from the second page, and moved the remaining two options (choosing to import into product list or inventory, and whether to reset stock of missing products) into the upload page at the start of the process.

#### What should we test?

Import options section removed. Choosing inventory/product list and whether to reset stock for absent products happens on the upload page. Resetting stock on missing products should be working correctly.

#### Release notes

Product Import: Simplified import settings and UI

Changelog Category: Changed

### CODE REVIEW

File: app/models/product_import/entry_processor.rb [hunk 1]
```diff
@@ -227,8 +233,8 @@ def permission_by_id?(supplier_id)
       @editable_enterprises.value?(Integer(supplier_id))
     end
 
-    def import_into_inventory_by_supplier?(supplier_id)
-      @import_settings[:settings] && @import_settings[:settings][supplier_id.to_s] && @import_settings[:settings][supplier_id.to_s]['import_into'] == 'inventories'
+    def importing_into_inventory?
+      @import_settings[:settings] && @import_settings[:settings]['import_into'] == 'inventories'
```
Comment (sauloperez) [inline app/models/product_import/entry_processor.rb]: not for this PR, but I think it's time to move the hash keys `settings, enterprise_to_reset and updated_ids` as ivars of this class specified in the constructor. Your fingers must hurt after typing all these hashes :trollface: 

As a result, `EntryProcessor` won't depend on the data structure of `@import_settings` anymore and that is a big win. If we happen to change it in the future (chances are that we will), it won't break this class.

Comment (sauloperez) [reply app/models/product_import/entry_processor.rb]: Now I see this class already has tons of ivars. That is a code :nose: . Before doing what I suggested, we should understand if some of these vars are hiding one or more concepts that deserve their own object.

**Comment by mkllnk** [comment]
@Matt-Yorkley I resolved a merge conflict to get this to test ready.

**Comment by Matt-Yorkley** [comment]
Staging now on: https://staging.openfoodnetwork.org.uk

**Comment by Matt-Yorkley** [comment]
Rebased, resolved, and redeployed to: https://staging.openfoodnetwork.org.uk

**Comment by RachL** [comment]
This is good to go for me :)
Here are my testing notes : 
https://docs.google.com/document/d/1zrErFKvIzkBfzMQZTZY_NUo5wBSZrsT_Xy7L7O8IVzA/edit#

As PR were mainly on UX issues so far I've tested only with not much more than 10 products in an upload file. If there are real test cases with much more products, I'm interested to access those file, as I might not be testing all weird / real-life cases that exists. ping @kirstenalarsen @myriamboure

**Comment by myriamboure** [comment]
Awesome, this is going so fast, what an awesome team! @RachL you can probably export a report with products and variants from openfoodfrance report section if you want to make tests with real life databases and huge files ;-)

**Pull request merged**